From: Caleb Connolly <caleb.connolly@linaro.org>
Date: Tue, 30 Nov 2021 23:44:48 +0000
Subject: [PATCH 36/79] WIP: arm64: dts: qcom: sdm845-xiaomi-beryllium-ebbg:
 fixup supplies

Fix the lab and ibb supplies, they should be on the display panel not
the touch screen!

also fix the touch screen compatible screen to include the TS model.
---
 .../boot/dts/qcom/sdm845-xiaomi-beryllium-ebbg.dts | 31 ++++++++++++++++------
 1 file changed, 23 insertions(+), 8 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-ebbg.dts b/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-ebbg.dts
index c7518b7..1386855 100644
--- a/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-ebbg.dts
+++ b/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-ebbg.dts
@@ -4,6 +4,19 @@
 
 #include "sdm845-xiaomi-beryllium-common.dtsi"
 
+/ {
+	/*
+	 * I don't know where the heck this regulator comes from
+	 * but it powers the touchscreen vdd rail (vsn/vsp??)
+	 */
+	ts_vdd: ts-vdd-regulator {
+		compatible = "regulator-fixed";
+		regulator-name = "ts_vdd";
+		regulator-min-microvolt = <3000000>;
+		regulator-max-microvolt = <3000000>;
+	};
+};
+
 &tlmm {
 
 	sde_dsi_active: sde_dsi_active {
@@ -28,7 +41,7 @@ mux {
 		config {
 			pins = "gpio6", "gpio52";
 			drive-strength = <2>;   /* 2 mA */
-			bias-pull-down;         /* PULL DOWN */
+			bias-pull-down;		/* PULL DOWN */
 		};
 	};
 };
@@ -36,15 +49,18 @@ config {
 
 &display_panel {
 
-        status = "okay";
+	status = "okay";
 
 	reset-gpios = <&tlmm 6 GPIO_ACTIVE_HIGH>;
 
+	lab-supply = <&lab>;
+	ibb-supply = <&ibb>;
+
 	pinctrl-names = "panel_active", "panel_suspend";
 	pinctrl-0 = <&sde_dsi_active>;
 	pinctrl-1 = <&sde_dsi_suspend>;
 
-        compatible = "ebbg,ft8719";
+	compatible = "ebbg,ft8719";
 
 };
 
@@ -120,15 +136,14 @@ touchscreen: focaltech@38 {
 		reg = <0x38>;
 		interrupt-parent = <&tlmm>;
 		interrupts = <31 0>;
-		vddio-supply = <&vreg_l14a_1p8>;
-		lab-supply = <&lab>;
-		ibb-supply = <&ibb>;
+		vdd-supply = <&ts_vdd>;
+		vcc-i2c-supply = <&vreg_l14a_1p8>;
 
 		focaltech,reset-gpio = <&tlmm 32 0x01>;
 		focaltech,irq-gpio = <&tlmm 31 0x02>;
 		focaltech,max-touch-number = <10>;
-		focaltech,display-coords = <0 0 1080 2246>;
-		pinctrl-names = "pmx_ts_active", "pmx_ts_suspend";
+		focaltech,display-size = <1080 2246>;
+		pinctrl-names = "ts_active", "ts_suspend";
 		pinctrl-0 = <&ts_int_active &ts_reset_active>;
 		pinctrl-1 = <&ts_int_suspend &ts_reset_suspend>;
 	};
